**Project 3: Profiling and modeling the colorectal cancer microbiome.**

* Trainer: **Alessio Milanese**, [Sunagawa Lab - Microbiome Research, ETH Zurich](https://micro.biol.ethz.ch/research/sunagawa.html.) and [SIB](https://www.sib.swiss/scientific-community/research/groups-and-group-leaders/10277-shinichi-sunagawa-s-group). [NCCR Microbiomes](https://nccr-microbiomes.ch/) member;
* Trainer: **Lukas Malfertheiner**, [Von Mering Lab - System Biology, University of Zurich](https://www.mls.uzh.ch/en/research/von-mering) and [SIB](https://www.sib.swiss/christian-von-mering-group). [NCCR Microbiomes](https://nccr-microbiomes.ch/) member;
* Description: Explore different cross-sectional metagenomic studies related to colorectal cancer (CRC), starting from taxonomic profiling to modeling by machine learning. Students will learn to work with metagenomic sequencing data and how to create taxonomic profiles from metagenomic samples using mOTUs [(Milanese et al. 2019)](https://www.nature.com/articles/s41467-019-08844-4). They will use R to explore properties typical for microbiome data (e.g., sparsity and sample size differences), use dimensionality reduction techniques and quantify within- and between-sample diversity. To explore associations between microbes and disease status, students will learn to use the SIAMCAT tool box [(Wirbel et al. 2021)](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-021-02306-1) and develop a model for CRC classification based on taxonomic profiles.

The project requires a basic understanding of the Unix command line and basic programming knowledge in R.
